skimage/restoration/inpaint.py

```diff
@@ -124,27 +254,75 @@ def inpaint_biharmonic(image, mask, multichannel=False):
         raise TypeError('Masked arrays are not supported')
 
     image = skimage.img_as_float(image)
-    mask = mask.astype(bool)
-
-    # Split inpainting mask into independent regions
-    kernel = ndi.morphology.generate_binary_structure(mask.ndim, 1)
-    mask_dilated = ndi.morphology.binary_dilation(mask, structure=kernel)
-    mask_labeled, num_labels = label(mask_dilated, return_num=True)
-    mask_labeled *= mask
+    mask = mask.astype(bool, copy=False)
 
     if not multichannel:
         image = image[..., np.newaxis]
-
     out = np.copy(image)
 
-    for idx_channel in range(image.shape[-1]):
-        known_points = image[..., idx_channel][~mask]
-        limits = (np.min(known_points), np.max(known_points))
-
-        for idx_region in range(1, num_labels+1):
-            mask_region = mask_labeled == idx_region
-            _inpaint_biharmonic_single_channel(mask_region,
-                out[..., idx_channel], limits)
+    # Create biharmonic coefficients ndarray
+    radius = 2
+    coef_shape = (2 * radius + 1,) * mask.ndim
+    coef_center = (radius,) * mask.ndim
+    neigh_coef_full, coef_idx, coef_vals = _get_neigh_coef(coef_shape,
+                                                           coef_center,
+                                                           dtype=out.dtype)
+
+    # stride for the last spatial dimension
+    channel_stride_bytes = out.strides[-2]
+
+    # offsets to all neighboring non-zero elements in the footprint
+    offsets = coef_idx - radius
+
+    # determine per-channel intensity limits
+    known_points = image[~mask]
+    limits = (known_points.min(axis=0), known_points.max(axis=0))
+
+    if split_into_regions:
+        # Split inpainting mask into independent regions
+        kernel = ndi.generate_binary_structure(mask.ndim, 1)
+        mask_dilated = ndi.binary_dilation(mask, structure=kernel)
+        mask_labeled, num_labels = label(mask_dilated, return_num=True)
```

Comment:
`num_labels` is no more used :tada: 
```suggestion
        mask_labeled = label(mask_dilated)
```

Reply:
Good catch. I also forgot to remove the `functools` and `operator` imports. will update it now